modules/videoio/src/cap_android_camera.cpp

```diff
@@ -308,12 +308,12 @@ class AndroidCameraCapture : public IVideoCapture
         AImage_getPlaneData(image.get(), 2, &vPixel, &vLen);
         AImage_getPlanePixelStride(image.get(), 1, &uvPixelStride);
 
-        if ( (uvPixelStride == 2) && (uPixel == vPixel + 1) && (yLen == frameWidth * frameHeight) && (uLen == ((yLen / 2) - 1)) && (vLen == uLen) ) {
+        if ( (uvPixelStride == 2) && (uPixel == vPixel + 1) && (yLen >= frameWidth * frameHeight) && (uLen >= ((yLen / 2) - 1)) && (vLen == uLen) ) {
```

Comment:
The code below doesn't expect pads:

> `buffer.insert(buffer.end(), yPixel, yPixel + yLen);`
> `buffer.insert(buffer.end(), vPixel, vPixel + yLen / 2);`

`buffer` expects packed data without alignment pads. This is why we have `yLen == frameWidth * frameHeight` check.
Relaxed check required data repacking.

The final result is corrupted (1+px shift per line). Try to capture vertical black/white lines (bands). And save it into file (e.g. through imwrite) for analysis.